submission/submitter/src/test/kotlin/ac/uk/ebi/biostd/submission/service/DoiServiceTest.kt

```diff
@@ -0,0 +1,366 @@
+import io.mockk.slot
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.core.io.FileSystemResource
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpHeaders.CONTENT_TYPE
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClient.RequestBodySpec
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.ZoneOffset.UTC
+import java.util.function.Consumer
+
+@ExtendWith(MockKExtension::class)
+class DoiServiceTest(
+    @MockK private val webClient: WebClient,
+    @MockK private val submitRequest: SubmitRequest,
+    @MockK private val previousVersion: ExtSubmission,
+) {
+    private val testInstance = DoiService(webClient, properties)
+    private val mockNow = OffsetDateTime.of(2020, 9, 21, 10, 11, 0, 0, UTC).toInstant()
+
+    @AfterEach
+    fun afterEach() = clearAllMocks()
+
+    @BeforeEach
+    fun beforeEach() {
+        mockkStatic(Instant::class)
+        every { Instant.now() } returns mockNow
+        every { submitRequest.previousVersion } returns null
+    }
+
+    @Test
+    fun `doi registration`(
+        @MockK requestSpec: RequestBodySpec,
+    ) {
+        val headersSlot = slot<Consumer<HttpHeaders>>()
+        val bodySlot = slot<LinkedMultiValueMap<String, Any>>()
+        val submission = submission {
+            title = "Test Submission"
+            attribute("DOI", "")
+
+            section("Study") {
+                section("Organization") {
+                    accNo = "o1"
+                    attribute("Name", "EMBL")
+                }
+
+                section("Author") {
+                    attribute("Name", "John Doe")
+                    attribute("ORCID", "12-32-45-82")
+                    attribute("Affiliation", "o1", ref = true)
+                }
+            }
+        }
+
+        every { submitRequest.submission } returns submission
+        every { webClient.post().uri(properties.endpoint) } returns requestSpec
+        every { requestSpec.bodyValue(capture(bodySlot)) } returns requestSpec
+        every { requestSpec.headers(capture(headersSlot)) } returns requestSpec
+        every { requestSpec.retrieve().bodyToMono(String::class.java).block() } returns "OK"
+
+        val doi = testInstance.calculateDoi(TEST_ACC_NO, submitRequest)
+        val body = bodySlot.captured
+        val headers = headersSlot.captured
+        val requestFile = body[FILE_PARAM]!!.first() as FileSystemResource
+        val expectedXml = Files.readString(Paths.get("src/test/resources/ExpectedDOIRequest.xml"))
```

Comment:
really subjective comment but I think test will be clear if you include expected xml at the end using Kotlin [multiline](https://kotlinlang.org/docs/java-to-kotlin-idioms-strings.html#use-multiline-strings) string.